Remove unused line height style for mce-content-body

kjellr opened the pull request:
`mce-content-body` is no longer applied to editable rich text elements. (I think that changed with #13697). As a result, this style is no longer picked up. 

Since this style is unused, there should be no visual difference in removing it. But here are screenshots to show where it used to show up back when it was active: 

## Before (WordPress 5.0.3)

![screen shot 2019-02-13 at 2 55 49 pm](https://user-images.githubusercontent.com/1202812/52739887-da487d00-2f9f-11e9-9045-9262eaa73779.png)

## Current (in `master`, and in this PR)

![screen shot 2019-02-13 at 2 54 51 pm](https://user-images.githubusercontent.com/1202812/52739805-b4bb7380-2f9f-11e9-81e2-423f8ca3e64e.png)


---

As a sidenote, applying that `mce-content-body` style to the `editor-styles-wrapper` was causing some theme style issues like #10067, so it's good to have it removed. ([More discussion here](https://github.com/WordPress/gutenberg/pull/13765#issuecomment-462773568)) 🙂

kjellr committed:
Remove unused line height style for mce-content-body

jasmussen reviewed (approved):
Nice one. This should be a trivial code cleanup since that style isn't used at all anymore.

However, can you go one step further now that you're at it, and also remove this line? 

https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/components/rich-text/style.scss#L10

I tried removing it, and the "line-height: inherit" makes sure that the correct line height is inherited from the editor styles wrapper. Which is superior. So might as well kill both while we're at it, right?

Looks good to me:

<img width="754" alt="screenshot 2019-02-14 at 08 03 30" src="https://user-images.githubusercontent.com/1204802/52768949-34445380-302f-11e9-9269-4384d502d1a4.png">

youknowriad commented:
@jasmussen This second line also applies to lists, quotes... I wonder if it has an impact there.

jasmussen commented:
Really solid thought, Riad. Looks to me like those are fine, and inherit correctly. So really, removing that line is just another feather in the cap for reducing editor style specificity!

<img width="625" alt="screenshot 2019-02-14 at 09 00 03" src="https://user-images.githubusercontent.com/1204802/52771788-21358180-3037-11e9-9999-f40a02ff3a35.png">

<img width="722" alt="screenshot 2019-02-14 at 09 00 51" src="https://user-images.githubusercontent.com/1204802/52771791-2266ae80-3037-11e9-8474-b165574fc14a.png">

☝️ don't mind the dark style, fixing a bug. And those all inherit the correct lineheights.

youknowriad commented:
Thanks for testing @jasmussen 👍

iseulde commented:
Looks good to me! Just also dropping here that `.editor-rich-text__tinymce` changed to `.editor-rich-text__editable`. Please avoid using this selector at all costs!

kjellr committed:
Remove unnecessary line-height property for .editor-rich-text-editable

This line height can be inherited with no negative effects. Removing this allows for less specific rules, making editor styles a bit easier.

kjellr commented:
Thanks, everyone!

> However, can you go one step further now that you're at it, and also remove this line?
> 
> https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/components/rich-text/style.scss#L10

Yep! I removed this in eafac98 and haven't noticed any issues either. I'll go ahead and merge once the tests are finished up. 👍
